### PR TITLE
Pubkey use brainpoolP320r1

### PR DESCRIPTION
**Pull request by klikooo**

### CODE REVIEW

File: src/keys/key.cpp [hunk 1]
```diff
@@ -141,47 +167,84 @@ bool CKey::SetPrivKey(const CPrivKey &privkey, bool fCompressedIn) {
...
     assert(fValid);
-    secp256k1_pubkey pubkey;
-    size_t clen = 65;
+
     CPubKey result;
-    int ret = secp256k1_ec_pubkey_create(secp256k1_context_sign, &pubkey, begin());
-    assert(ret);
-    secp256k1_ec_pubkey_serialize(secp256k1_context_sign, (unsigned char*)result.begin(), &clen, &pubkey, fCompressed ? SECP256K1_EC_COMPRESSED : SECP256K1_EC_UNCOMPRESSED);
-    assert(result.size() == clen);
-    assert(result.IsValid());
```
Comment (jonathanraes) [inline src/keys/key.cpp:166]: asserts may be good to keep in there

**Review by jonathanraes — APPROVED** [review]

File: src/keys/key.cpp [hunk 2]
```diff
@@ -118,19 +120,43 @@ static int ec_privkey_export_der(const secp256k1_context *ctx, unsigned char *pr
...
 }
- 
+
 bool CKey::Check(const unsigned char *vch) {
-    return secp256k1_ec_seckey_verify(secp256k1_context_sign, vch);
+    //IS NOT USED
+    CryptoPP::AutoSeededRandomPool prng;
+    CryptoPP::ECDSA<CryptoPP::ECP, CryptoPP::SHA256>::PrivateKey privKey;
+
+    //privKey.Initialize( prng, CryptoPP::ASN1::brainpoolP320r1() );
```
Comment (wkmeijer) [inline src/keys/key.cpp]: commented code

File: src/keys/key.cpp [hunk 3]
```diff
@@ -141,47 +167,83 @@ bool CKey::SetPrivKey(const CPrivKey &privkey, bool fCompressedIn) {
...
+
+    //Determine maximum size, allocate a string with 4 times the maximum size. The max size is 80 voor brainpoolP320r1
+    size_t siglen = signer.MaxSignatureLength();
+    std::string signature(siglen * 4, 0x00);
+    
+    //Sign the hash in 4 different parts
+    unsigned int i=0;
+    for(i=0; i< 4; i++ ) {
+        siglen = signer.SignMessage( prng, (const byte*) &hash + i*8, 8, (byte*)signature.data() + i* CRYPTOPP_SIGNATURE_SIZE );
+        //signature.resize(siglen);
```
Comment (wkmeijer) [inline src/keys/key.cpp]: commented code

File: src/script/interpreter.cpp [hunk 4]
```diff
@@ -187,14 +192,17 @@ bool static IsDefinedHashtypeSignature(const valtype &vchSig) {
 bool static CheckSignatureEncoding(const valtype &vchSig, unsigned int flags, ScriptError* serror) {
     // Empty signature. Not strictly DER encoded, but allowed to provide a
     // compact way to provide an invalid signature for use with CHECK(MULTI)SIG
+    //return true;
```
Comment (wkmeijer) [inline src/script/interpreter.cpp]: commented code

File: src/script/interpreter.cpp [hunk 5]
```diff
@@ -187,14 +192,17 @@ bool static IsDefinedHashtypeSignature(const valtype &vchSig) {
...
     }
     if ((flags & (SCRIPT_VERIFY_DERSIG | SCRIPT_VERIFY_LOW_S | SCRIPT_VERIFY_STRICTENC)) != 0 && !IsValidSignatureEncoding(vchSig)) {
         return set_error(serror, SCRIPT_ERR_SIG_DER);
-    } else if ((flags & SCRIPT_VERIFY_LOW_S) != 0 && !IsLowDERSignature(vchSig, serror)) {
+        
+    //there is no low DER signature right now
+    /*} else if ((flags & SCRIPT_VERIFY_LOW_S) != 0 && !IsLowDERSignature(vchSig, serror)) {
         // serror is set
-        return false;
+        return false;*/
```
Comment (wkmeijer) [inline src/script/interpreter.cpp]: commented code

**Review by wkmeijer — CHANGES_REQUESTED** [review]
Since I can't really say much about the functionality of the code because of my lack of Multichain knowledge. I mainly checked for code style and good practice.

I found 4 pieces of commented code that according to me can be removed.

**Comment by klikooo** [comment]
Removed the comments, some of the comments are old MultiChain code which I think should stay there

**Review by wkmeijer — APPROVED** [review]

**Comment by klikooo** [comment]
Changing to develop branch instead of master

**Pull request merged**